authz/rbac_translator.go

```diff
@@ -199,26 +198,20 @@ func parseHeaderValues(key string, values []string) []*v3rbacpb.Permission {
 	return vs
 }
 
-// hop-by-hop headers.
-var hopHeaders = []string{
-	"Connection",
-	"Keep-Alive",
-	"Proxy-Authenticate",
-	"Proxy-Authorization",
-	"Te",
-	"Trailer",
-	"Transfer-Encoding",
-	"Upgrade",
-}
-
 func unsupportedHeader(key string) bool {
-	if strings.HasPrefix(key, ":") || strings.HasPrefix(key, "grpc-") || key == "host" || key == "Host" {
+	switch {
+	case strings.HasPrefix(key, ":"):
+		return true
+	case strings.HasPrefix(key, "grpc-"):
+		return true
+	case key == "host":
+		return true
+	// hop-by-hop headers.
+	case key == "connection" || key == "keep-alive" ||
+		key == "proxy-authenticate" || key == "proxy-authorization" ||
+		key == "te" || key == "trailer" || key == "transfer-encoding" ||
+		key == "upgrade":
```

Comment:
The `map` idea actually is probably better (performance-wise and visually) now that I look at this, sorry.  Usually a `switch` is just as good or better than a map, because the compiler can use a jump table, but for strings I don't think that's true.  E.g.

```go
unsupportedHeaders := map[string]bool{
	"host": true, "connection": true, "keep-alive": true,
	.....
}
func unsupportedHeader(key string) bool {
	return strings.HasPrefix(key, ":") || strings.HasPrefix(key, "grpc-") || unsupportedHeaders[key]
```

Reply:
Sure. This looks much better. Updated code. Thank you!